packages/flutter/lib/src/widgets/scroll_physics.dart

```diff
@@ -969,4 +972,7 @@ class NeverScrollableScrollPhysics extends ScrollPhysics {
 
   @override
   bool get allowImplicitScrolling => false;
+
+  @override
+  bool get allowUserScrolling => false;
```

Comment:
Doesn't `shouldAcceptUserOffset` above provide the same thing?

Reply:
Hm, the default implementation of shouldAcceptUserOffset might not work as this is intended. However, I think if adding this allowUserScrolling is the right way to resolve this, it should work with shouldAcceptUserOffset, or at least document what the difference between them is. Should shouldAcceptUserOffset also consider allowUserScrolling?

Reply:
@Piinks Nice point! Let us review the original issue again : )
1, The framework warp a gesture recognizer only if `physics.shouldAcceptUserOffset(this)` returns true.
https://github.com/flutter/flutter/blob/f266cccdeadb9a5fe7ceed0c4d98a68cf6f27879/packages/flutter/lib/src/widgets/scrollable.dart#L645-L649
So, If the DSS's extent becomes large enough, the scrollable will become unscrollable and the gesture will be removed (The extent's size is actually the scrollable viewport). Then the DSS is stuck and cannot be pulled back. This is the #31739 root cause.

2, Here is `_DraggableScrollableSheetScrollPosition.applyUserOffset` 's implement
https://github.com/flutter/flutter/blob/f266cccdeadb9a5fe7ceed0c4d98a68cf6f27879/packages/flutter/lib/src/widgets/draggable_scrollable_sheet.dart#L877-L882

It set the extent's size by the user scrolling. So, if we want fix the original issue we must let the scrollable always scrollable, right?

Regardless of whether #31832 or #109081, they essentially make the scrollable to be always scrollable, but #31832 is a bit hack and introduces issue #98314. 

Unfortunately, #109081 introduces issue #121021 because it ignores the ‘NeverScrollableScrollPhysics’ case. This is why I make this change to introduce a method to distinguish the `NeverScrollableScrollPhysics` and others.

In addition, I tried other solutions at the same time, such as caching the old physics, and then applying it to the user scrolling, but I gave up for it is a bit complicated.

Glad to here your thoughts : )

Reply:
Ah thank you for this! I really appreciate the context. I came across https://github.com/flutter/flutter/issues/113753 which is a similar-ish case of the explicit physics being overridden by a widget applying other physics.  

This makes me think, what do you think about some way for users to specify that physics should not be meddled with? And then maybe in `applyTo` we check that? I dunno, just brainstorming..
However, this is slightly different from #113753 in that in the nested scroll view, we see  userSuppliedPhysics.applyTo(extraImplementationPhysics), while here we see extraImplementationPhysics.applyTo(userSuppliedPhysics).. so this idea may not actually work...

Since it appears there is a pattern where the user may not want us to to change the scroll physics (I am sure there are other places where we do this, overriding the value explicitly set by the user), what are you thoughts on approaching this in a way that lets users preserve their preferred scroll physics? (not exclusive to just NeverScrollable and `allowUserScrolling`)

Reply:
`Ah thank you for this! I really appreciate the context. I came across https://github.com/flutter/flutter/issues/113753 which is a similar-ish case of the explicit physics being overridden by a widget applying other physics.`

It seems like physics overridden is not the root cause of that issue but this patch may help to fix that : )

`This makes me think, what do you think about some way for users to specify that physics should not be meddled with?`

Good idea but I'm concerned that doing this would be a breaking change?